wandb/apis/public.py

```diff
@@ -606,7 +606,7 @@ def artifact_versions(self, type_name, name, per_page=50):
         return artifact_type.collection(collection_name).versions(per_page=per_page)
 
     @normalize_exceptions
-    def artifact(self, name, type=None):
+    def artifact(self, name, type=None, use_as=None):
```

Comment:
What is the case when someone would need this param? i think it is:
```
art = api.artifact(..., use_as=X)
run.use_artifact(art)
```
is that correct? In that case, couldn't they just do
```
art = api.artifact(...)
run.use_artifact(art, use_as=X)
```
I think this would reduce the amount of `use_as` passing around that is happening

Reply:
Good call, I'll do that

Reply:
Oh wait, this is actually being used by use_artifact to set the value in the Artifact class. I can make this private as well